tests/unit/contrib/compilation_infra.py

```diff
@@ -1004,6 +1004,104 @@ def inputs_func(iter):
         )
 
 
+def partition_pad_conv2d_pool2d_nhwc_oihw_test(
+    in_shape,
+    padding,
+    pad_value,
+    w_shape,
+    conv_padding,
+    conv_strides,
+    conv_dilation,
+    pool_type,
+    pool_size,
+    pool_padding=[0, 0],
+    pool_strides=[1, 1],
+    conv_groups=1,
+    conv_invalid=False,
+    kernel_layout="OIHW",
+    targets=["DPUCZDX8G-zcu104"],
+) -> None:
+
+    for target in targets:
+        xgraph = _create_pad_conv2d_pool2d_nhwc_oihw(
+            in_shape,
+            padding,
+            pad_value,
+            w_shape,
+            conv_padding,
+            conv_strides,
+            conv_dilation,
+            pool_type,
+            pool_size,
+            pool_padding,
+            pool_strides,
+            conv_groups,
+            conv_invalid,
+            kernel_layout,
+            target,
+        )
+
+        assert xgraph.get_layers()[1].type.to_list()[0] == 'Pad'
+        assert xgraph.get_layers()[1].target == 'cpu'
+        assert xgraph.get_layers()[2].type.to_list()[0] == 'Convolution'
+        assert xgraph.get_layers()[2].target == target
+        assert xgraph.get_layers()[3].type.to_list()[0] == 'Pooling'
+        assert xgraph.get_layers()[3].target == target
+
+
+
+def _create_pad_conv2d_pool2d_nhwc_oihw(
```

Comment:
Could you move this function before https://github.com/Xilinx/pyxir/pull/77/files#diff-81112a007a7d11d0e684f8e307a0215af0481be9a64f800e08ad2342eb944ce3R1007? This actually works but declaring functions before they are being used is good convention for easier look and no risk on encountering an error with the function not being declared here: https://github.com/Xilinx/pyxir/pull/77/files#diff-81112a007a7d11d0e684f8e307a0215af0481be9a64f800e08ad2342eb944ce3R1026